url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/logic/ItemLogic.java

```diff
@@ -0,0 +1,622 @@
+import net.nexon.vindictus.itemcomparer.modell.ext.Gloves;
+import net.nexon.vindictus.itemcomparer.modell.ext.Helm;
+import net.nexon.vindictus.itemcomparer.modell.ext.Pants;
+import net.nexon.vindictus.itemcomparer.modell.ext.Scroll_Pre;
+import net.nexon.vindictus.itemcomparer.modell.ext.Scroll_Suf;
+import net.nexon.vindictus.itemcomparer.modell.ext.Shoes;
+
+public class ItemLogic {
+
+	private HashSet<Shoes> shoes_changed = new HashSet<>();
+	private HashSet<Pants> pants_changed = new HashSet<>();
+	private HashSet<Gloves> gloves_changed = new HashSet<>();
+	private HashSet<Armor> armors_changed = new HashSet<>();
+	private HashSet<Helm> helms_changed = new HashSet<>();
+
+	private void deleteChanged(Items items) {
+		items.getShoes().removeAll(shoes_changed);
+		items.getPants().removeAll(pants_changed);
+		items.getArmors().removeAll(armors_changed);
+		items.getGloves().removeAll(gloves_changed);
+		items.getHelms().removeAll(helms_changed);
+
+	}
+
+	public void duplicateItems(Items items, String prefix, String suffix) {
+
+		if (prefix == null && suffix == null) {
+			return;
+		}
+
+		prefix = prefix.trim();
+		suffix = suffix.trim();
+
+		if (prefix.length() < 1 && suffix.length() < 1) {
+			return;
+		}
+
+		List<Scroll_Pre> pres = new ArrayList<>();
+		List<Scroll_Suf> sufs = new ArrayList<>();
+
+		String[] prefixes = prefix.split(",");
+		for (String x : prefixes) {
+			try {
+				pres.add(StaticScrolls.searchPre(x.trim()));
+			} catch (Exception e) {
+			}
+		}
+
+		String[] suffixes = suffix.split(",");
+		for (String x : suffixes) {
+			try {
+				sufs.add(StaticScrolls.searchSuf(x.trim()));
+			} catch (Exception e) {
+			}
+		}
+
+		duplicateItems(items, false, pres, sufs);
+	}
+
+	private void duplicateItems(Items items, boolean keepChanged,
+			List<Scroll_Pre> pres, List<Scroll_Suf> sufs) {
+
+		List<Item> newitems = new ArrayList<>();
+		System.out.println(pres + " " + sufs);
+
+		if (pres.size() > 0 && sufs.size() > 0) {
+			for (Scroll_Pre pre : pres) {
+				for (Scroll_Suf suf : sufs) {
+					newitems.addAll(addScrollsToItems(items, pre, suf));
+				}
+			}
+		}
+
+		if (pres.size() > 0 && sufs.size() == 0) {
+			for (Scroll_Pre pre : pres) {
+				newitems.addAll(addScrollsToItems(items, pre, null));
+
+			}
+		}
+		if (pres.size() == 0 && sufs.size() > 0) {
+			for (Scroll_Suf suf : sufs) {
+				newitems.addAll(addScrollsToItems(items, null, suf));
+			}
+
+		}
+
+		if (!keepChanged) {
+			deleteChanged(items);
+		}
+
+		for (Item x : newitems) {
+			if (x instanceof Shoes) {
+				items.getShoes().add((Shoes) x);
+			}
+			if (x instanceof Pants) {
+				items.getPants().add((Pants) x);
+			}
+			if (x instanceof Armor) {
+				items.getArmors().add((Armor) x);
+			}
+			if (x instanceof Gloves) {
+				items.getGloves().add((Gloves) x);
+			}
+			if (x instanceof Helm) {
+				items.getHelms().add((Helm) x);
+			}
+
+		}
+
+	}
+
+	private List<Item> addScrollsToItems(Items items, Scroll_Pre newpre,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 135-142, lines 143-150, lines 151-158, lines 159-166, lines 167-174 into separate methods.